packages/flutter/lib/src/material/navigation_rail.dart

```diff
@@ -606,6 +653,20 @@ class _RailDestination extends StatelessWidget {
         final double verticalPadding = lerpDouble(_verticalDestinationPaddingNoLabel, _verticalDestinationPaddingWithLabel, appearingAnimationValue)!;
         final Interval interval = selected ? const Interval(0.25, 0.75) : const Interval(0.75, 1.0);
         final Animation<double> labelFadeAnimation = destinationAnimation.drive(CurveTween(curve: interval));
+
+        final Widget iconWithIndicator =  Stack(
```

Comment:
It would be better to avoid building the Stack if useIndicator is false, since Stack has a non-trivial cost.

Since this subtree appears twice, maybe create a private widget that DTRT?

Reply:
Good idea - done in this commit: https://github.com/flutter/flutter/pull/92930/commits/a700a4831fa78025ca80acc17eec9bc78b43a0e2